procedures/statement_performance_analyzer.sql

```diff
@@ -0,0 +1,792 @@
+
+             mysql> CALL sys.statement_performance_analyzer(''create_table'', ''mydb.digests_prev'', NULL);
+             Query OK, 0 rows affected (0.10 sec)
+
+             shell$ watch -n 60 "mysql sys --table -e \"
+             > SET @sys.statement_analyzer_view = ''mydb.my_statements'';
+             > SET @sys.statement_analyzer_limit = 10;
+             > CALL statement_performance_analyzer(''snapshot'', NULL, NULL);
+             > CALL statement_performance_analyzer(''delta'', ''mydb.digests_prev'', ''custom'');
+             > CALL statement_performance_analyzer(''save'', ''mydb.digests_prev'', NULL);
+             > \""
+
+             Every 60.0s: mysql sys --table -e "                                                                                                   ...  Mon Dec 22 10:58:51 2014
+
+             +----------------------------------+
+             | Next Output                      |
+             +----------------------------------+
+             | Top 10 Queries Using Custom View |
+             +----------------------------------+
+             +-------------------+-------+------------+---------------+-------------+---------------+-------------------+-------------------+----------------------------------+
+             | query             | db    | exec_count | total_latency | avg_latency | rows_sent_avg | rows_examined_avg | rows_affected_avg | digest                           |
+             +-------------------+-------+------------+---------------+-------------+---------------+-------------------+-------------------+----------------------------------+
+             ...
+            '
+    SQL SECURITY INVOKER
+    NOT DETERMINISTIC
+    CONTAINS SQL
+BEGIN
+    DECLARE v_table_exists, v_tmp_digests_table_exists, v_custom_view_exists ENUM('', 'BASE TABLE', 'VIEW', 'TEMPORARY') DEFAULT '';
+    DECLARE v_this_thread_enabled ENUM('YES', 'NO');
+    DECLARE v_force_new_snapshot BOOLEAN DEFAULT FALSE;
+    DECLARE v_digests_table VARCHAR(133);
+    DECLARE v_quoted_table, v_quoted_custom_view VARCHAR(133) DEFAULT '';
+    DECLARE v_table_db, v_table_name, v_custom_db, v_custom_name VARCHAR(64);
+    DECLARE v_digest_table_template, v_checksum_ref, v_checksum_table text;
+    -- Maximum supported length for MESSAGE_TEXT with the SIGNAL command is 128 chars.
+    DECLARE v_error_msg VARCHAR(128);
+
+
+    -- Don't instrument this thread
+    SELECT INSTRUMENTED INTO v_this_thread_enabled FROM performance_schema.threads WHERE PROCESSLIST_ID = CONNECTION_ID();
+    CALL sys.ps_setup_disable_thread(CONNECTION_ID());
+
+    -- Temporary table are used - disable sql_log_bin if necessary to prevent them replicating
+    SET @log_bin := @@sql_log_bin;
+    IF (@log_bin = 1) THEN
+        SET sql_log_bin = 0;
+    END IF;
+
+
+    -- Set configuration options
+    IF (@sys.statement_analyzer_limit IS NULL) THEN
+        SET @sys.statement_analyzer_limit = sys.sys_get_config('statement_analyzer_limit', '100');
```

Comment:
This should be consistent like the diagnostics routine variables - i.e it should prefix with the routine name like:

statement_performance_analyzer.limit